test/index.test.js

```diff
@@ -26,20 +26,35 @@ describe('RxRouter', () => {
 
   it('should support multiple routes, methods and Epics', async () => {
     const router = new RxRouter()
+    const methods = [
+      'del',
+      'delete',
+      'get',
+      'patch',
+      'post',
+      'put'
+    ];
 
-    router.get('/test1', obs => obs.mapTo('get-test1'))
-    router.post('/test1', obs => obs.mapTo('post-test1'))
-    router.put('/test2', obs => obs.mapTo('put-test2'))
-    router.del('/test3', obs => obs.mapTo('del-test3'))
-    router.delete('/test4', obs => obs.mapTo('del-test4'))
+    methods.forEach((methodName) => {
+      router[methodName](`/${methodName}`, obs => obs.mapTo(methodName))
+    });
 
     const request = init(router)
 
-    await request.get('/test1').expect('get-test1')
-    await request.post('/test1').expect('post-test1')
-    await request.put('/test2').expect('put-test2')
-    await request.del('/test3').expect('del-test3')
-    await request.delete('/test4').expect('del-test4')
+    for (let methodName of methods) {
+      await request[methodName](`/${methodName}`).expect(methodName)
+    }
+  })
+
+  it('does not replace route paths if exists', async () => {
```

Comment:
This is an interesting test and I think there should be even more tests covering subscribing to the `request` Subject or using the `response` Subject independently (they are returned by router.get/post etc.).

But this test could also be misleading, because I think the response depends on which Observable completes first. I'd like to remove this test from this PR and deal with that issue separately.